plutus-tx/src/PlutusTx/Builtins.hs

```diff
@@ -453,14 +457,14 @@ matchData
     -> (Integer -> r)
     -> (BuiltinByteString -> r)
     -> r
-matchData d constrCase mapCase listCase iCase bCase =
+matchData ~d ~constrCase ~mapCase ~listCase ~iCase ~bCase =
    chooseData
    d
-   (\_ -> uncurry constrCase (unsafeDataAsConstr d))
-   (\_ -> mapCase (unsafeDataAsMap d))
-   (\_ -> listCase (unsafeDataAsList d))
-   (\_ -> iCase (unsafeDataAsI d))
-   (\_ -> bCase (unsafeDataAsB d))
+   (\(~_) -> uncurry constrCase (unsafeDataAsConstr d))
```

Comment:
This should perhaps be a `forall` anyway here? So that it compiles to a `delay`. Not sure. @michaelpj

Reply:
Yeah maybe. I also think these tildes cannot do anything.